cluster/manager/manager.go

```diff
@@ -327,27 +329,78 @@ func (c *ClusterManager) UpdateSchedulerNodeName(schedulerNodeName string) error
 	defer c.selfNodeLock.Unlock()
 	c.selfNode.SchedulerNodeName = schedulerNodeName
 
-	kvdb := kvdb.Instance()
-	kvlock, err := kvdb.LockWithID(clusterLockKey, c.selfNode.Id)
-	if err != nil {
-		logrus.Warnln("Unable to obtain cluster lock for updating config", err)
-		return err
+	updateCallbackFn := func(db *cluster.ClusterInfo) (bool, error) {
+		nodeEntry, ok := db.NodeEntries[c.selfNode.Id]
+		if !ok {
+			return false, fmt.Errorf("Node not found in cluster database")
+		}
+		nodeEntry.SchedulerNodeName = schedulerNodeName
+		db.NodeEntries[c.selfNode.Id] = nodeEntry
+		return true, nil
 	}
-	defer kvdb.Unlock(kvlock)
 
-	db, _, err := readClusterInfo()
-	if err != nil {
-		return err
+	return updateLockedDB("update-scheduler-name", c.selfNode.Id, updateCallbackFn)
+}
+
+func (c *ClusterManager) DeactivateClusterDomain(deactivateRequest *api.DeactivateClusterDomainRequest) error {
+	c.selfNodeLock.Lock()
+	defer c.selfNodeLock.Unlock()
+
+	updateCallbackFn := func(db *cluster.ClusterInfo) (bool, error) {
+		isActive, _ := db.ClusterDomains.ActiveMap[deactivateRequest.ClusterDomain]
+		if !isActive {
+			// Cluster domain is not active. No need of update
+			return false, nil
+		}
+		db.ClusterDomains.ActiveMap[deactivateRequest.ClusterDomain] = false
+		return true, nil
 	}
+	return updateLockedDB("deactivate-cluster-domain", c.selfNode.Id, updateCallbackFn)
+}
 
-	nodeEntry, ok := db.NodeEntries[c.selfNode.Id]
-	if !ok {
-		return fmt.Errorf("Node not found in cluster database")
+func (c *ClusterManager) ActivateClusterDomain(activateRequest *api.ActivateClusterDomainRequest) error {
+	c.selfNodeLock.Lock()
+	defer c.selfNodeLock.Unlock()
+
+	updateCallbackFn := func(db *cluster.ClusterInfo) (bool, error) {
+		isActive, _ := db.ClusterDomains.ActiveMap[activateRequest.ClusterDomain]
```

Comment:
if !db.ClusterDomains.ActiveMap[activateRequest.ClusterDomain] {
  db.ClusterDomains.ActiveMap[activateRequest.ClusterDomain]  = true  
 return true, nil
}
return false, nil
may be shorter.